File: src/api/jobPosting.ts
// ============================================================
// 招聘数据聚合 API（/jobs 页面）
// ============================================================

import { apiClient } from './client'
import type { Pagination } from './types'

export interface JobPosting {
  id: string
  source: string
  sourceId?: string
  companyName: string
  industry?: string
  industryCategory?: string // 归一化行业大类（配色/筛选用）
  recruitmentType?: string
  recruitmentCategory?: string // 归一化招聘类型大类（配色/筛选用）
  openDate?: string // ISO 日期字符串
  location?: string
  positions?: string
  applicationUrl?: string
  referralCode?: string
  notes?: string
  isActive: boolean
  createdAt: string
  updatedAt: string
  scrapedAt: string
}

export interface JobFilters {
  industries: string[]
  types: string[]
}

export interface SyncResult {
  total: number
  inserted: number
  updated: number
  deactivated: number
  errors: number
  source?: string
  startedAt: string
  finishedAt: string
  durationMs: number
}

export interface JobListParams {
  industry?: string
  type?: string
  keyword?: string
  page?: number
  pageSize?: number
}

export interface JobPostingListResponse {
  items: JobPosting[]
  pagination: Pagination
}

export const jobPostingApi = {
  getJobPostings: (params: JobListParams = {}) =>
    apiClient.get<JobPostingListResponse>(`/job-postings?${buildQuery(params)}`),
  getJobFilters: () => apiClient.get<JobFilters>('/job-postings/filters'),
  syncJobPostings: () => apiClient.post<SyncResult>('/job-postings/sync', {}, { auth: true }),
}

function buildQuery(params: JobListParams): string {
  const qs = new URLSearchParams()
  if (params.industry) qs.set('industry', params.industry)
  if (params.type) qs.set('type', params.type)
  if (params.keyword) qs.set('keyword', params.keyword)
  if (params.page) qs.set('page', String(params.page))
  if (params.pageSize) qs.set('pageSize', String(params.pageSize))
  return qs.toString()
}

File: backend/internal/model/job_posting.go
package model

import "time"

// JobPosting 招聘数据聚合表（来自腾讯文档智能表格）
type JobPosting struct {
	ID                  string     `json:"id"`
	Source              string     `json:"source"`
	SourceID            string     `json:"sourceId,omitempty"`
	CompanyName         string     `json:"companyName"`
	Industry            string     `json:"industry,omitempty"`
	IndustryCategory    string     `json:"industryCategory,omitempty"` // 归一化行业大类（筛选用）
	RecruitmentType     string     `json:"recruitmentType,omitempty"`
	RecruitmentCategory string     `json:"recruitmentCategory,omitempty"` // 归一化招聘类型大类（筛选用）
	OpenDate            *time.Time `json:"openDate,omitempty"`
	Location            string     `json:"location,omitempty"`
	Positions           string     `json:"positions,omitempty"`
	ApplicationURL      string     `json:"applicationUrl,omitempty"`
	ReferralCode        string     `json:"referralCode,omitempty"`
	Notes               string     `json:"notes,omitempty"`
	IsActive            bool       `json:"isActive"`
	CreatedAt           time.Time  `json:"createdAt"`
	UpdatedAt           time.Time  `json:"updatedAt"`
	ScrapedAt           time.Time  `json:"scrapedAt"`
	Applied             bool       `json:"applied"` // 当前用户是否已标记「已投递」；未登录时始终为 false
}

// JobPostingFilters 列表查询筛选条件
type JobPostingFilters struct {
	Industry        string
	RecruitmentType string
	Keyword         string
	Applied         string // ""(不筛选) | "true"(已投递) | "false"(未投递)
	Sort            string // "open_date_desc"(默认) | "open_date_asc"
	Page            int
	PageSize        int
	UserID          string // 当前登录用户 ID，用于关联/筛选投递标记；未登录为空
}

// JobPostingListResponse 列表响应
type JobPostingListResponse struct {
	Items      []JobPosting `json:"items"`
	Pagination Pagination   `json:"pagination"`
}

// JobPostingFiltersResponse 公开筛选枚举值
type JobPostingFiltersResponse struct {
	Industries []string `json:"industries"`
	Types      []string `json:"types"`
}

// SetJobPostingMarkRequest 标记 / 取消标记「已投递」请求体
type SetJobPostingMarkRequest struct {
	Applied bool `json:"applied"`
}

// SyncResult 一次同步的统计结果
type SyncResult struct {
	Total       int    `json:"total"`
	Inserted    int    `json:"inserted"`
	Updated     int    `json:"updated"`
	Deactivated int    `json:"deactivated"`
	Errors      int    `json:"errors"`
	Source      string `json:"source,omitempty"`
	StartedAt   string `json:"startedAt"`
	FinishedAt  string `json:"finishedAt"`
	DurationMs  int64  `json:"durationMs"`
}

// UpsertJobPostingsResult UpsertJobPostings 的返回结果：计数 + 本次真正新插入的岗位明细
// （InsertedItems 用于上层将新增岗位追加到 Redis「最近新增」列表，仅统计新插入，更新的不重复推送）
type UpsertJobPostingsResult struct {
	Total         int
	Inserted      int
	Updated       int
	InsertedItems []NewJobItem
}
